mlflow/metrics/genai/model_utils.py

```diff
@@ -42,48 +41,45 @@ def _parse_model_uri(model_uri):
 
 def _call_openai_api(openai_uri, payload):
     """Wrapper around the OpenAI API to make it compatible with the MLflow Gateway API."""
-    from mlflow.gateway.config import RouteConfig
-    from mlflow.gateway.providers.openai import OpenAIProvider
 
     if "OPENAI_API_KEY" not in os.environ:
         raise MlflowException(
             "OPENAI_API_KEY environment variable not set",
             error_code=INVALID_PARAMETER_VALUE,
         )
 
-    config = {"openai_api_key": os.environ["OPENAI_API_KEY"]}
-    if "OPENAI_API_BASE" in os.environ:
-        config["openai_api_base"] = os.environ["OPENAI_API_BASE"]
-    if "OPENAI_API_TYPE" in os.environ:
-        config["openai_api_type"] = os.environ["OPENAI_API_TYPE"]
-    if "OPENAI_API_VERSION" in os.environ:
-        config["openai_api_version"] = os.environ["OPENAI_API_VERSION"]
-    if "OPENAI_DEPLOYMENT_NAME" in os.environ:
-        config["openai_deployment_name"] = os.environ["OPENAI_DEPLOYMENT_NAME"]
-
-    route_config = RouteConfig(
-        name="openai",
-        route_type=ROUTE_TYPE,
-        model={
-            "name": openai_uri,
-            "provider": "openai",
-            "config": config,
-        },
-    )
-    openai_provider = OpenAIProvider(route_config)
-
-    payload = openai_provider._prepare_completion_request_payload(payload)
-
     import openai
 
+    from mlflow.gateway.providers.utils import rename_payload_keys
+    from mlflow.openai import _get_api_config
     from mlflow.openai.api_request_parallel_processor import process_api_requests
     from mlflow.openai.utils import _OAITokenHolder
 
-    api_token = _OAITokenHolder(os.environ.get("OPENAI_API_TYPE", "openai"))
+    api_config = _get_api_config()
+    api_token = _OAITokenHolder(api_config.api_type)
+    envs = {
+        x: getattr(api_config, x)
+        for x in ["api_base", "api_version", "api_type", "engine", "deployment_id"]
+        if getattr(api_config, x) is not None
+    }
```

Comment:
Do we need all the other code from `_OpenAIWrapper` in this file`mlflow/openai/__init__.py`? 

```        api_type = self.model.get("api_type") or self.envs.get("api_type")
        if api_type in ("azure", "azure_ad", "azuread"):
            deployment_id = self.model.get("deployment_id") or self.envs.get("deployment_id")
            if self.model.get("engine") or self.envs.get("engine"):
                # Avoid using both parameters as they serve the same purpose
                # Invalid inputs:
                #   - Wrong engine + correct/wrong deployment_id
                #   - No engine + wrong deployment_id
                # Valid inputs:
                #   - Correct engine + correct/wrong deployment_id
                #   - No engine + correct deployment_id
                if deployment_id is not None:
                    _logger.warning(
                        "Both engine and deployment_id are set. "
                        "Using engine as it takes precedence."
                    )
            elif deployment_id is None:
                raise MlflowException(
                    "Either engine or deployment_id must be set for Azure OpenAI API",
                )

```